tidy3d/components/monitor.py

```diff
@@ -447,6 +383,247 @@ def storage_size(self, num_cells: int, tmesh: int) -> int:
+    ...     )
+    """
+
+    _data_type: Literal["Near2FarData"] = pydantic.Field("Near2FarAngleData")
+
+    theta: ObsGridArray = pydantic.Field(
+        ...,
+        title="Polar Angles",
+        description="Polar angles relative to ``local_origin`` at which to compute far fields.",
+        units=RADIAN,
+    )
+
+    phi: ObsGridArray = pydantic.Field(
+        ...,
+        title="Azimuth Angles",
+        description="Azimuth angles relative to ``local_origin`` at which to compute far fields.",
+        units=RADIAN,
+    )
+
+    def storage_size(self, num_cells: int, tmesh: ArrayLike[float, 1]) -> int:
+        # stores 1 complex number per pair of angles, per frequency,
+        # for N_theta, N_phi, L_theta, and L_phi (4 components)
+        return BYTES_COMPLEX * len(self.theta) * len(self.phi) * len(self.freqs) * 4
+
+
+class Near2FarCartesianMonitor(AbstractNear2FarMonitor):
+    """:class:`Monitor` that samples electromagnetic near fields in the frequency domain
+       and invokes the computation of far fields on a Cartesian observation plane.
+
+    Example
+    -------
+    >>> monitor = Near2FarCartesianMonitor(
+    ...     center=(1,2,3),
+    ...     size=(2,2,2),
+    ...     freqs=[250e12, 300e12],
+    ...     name='n2f_monitor',
+    ...     custom_origin=(1,2,3),
+    ...     x=[-1, 0, 1],
+    ...     y=[-2, -1, 0, 1, 2],
+    ...     plane_axis=2,
+    ...     plane_distance=5
+    ...     )
+    """
+
+    _data_type: Literal["Near2FarCartesianData"] = pydantic.Field("Near2FarCartesianData")
+
+    plane_axis: Axis = pydantic.Field(
+        ...,
+        title="Observation plane axis",
+        description="Axis along which the observation plane is oriented.",
+    )
+
+    plane_distance: float = pydantic.Field(
+        ...,
+        title="Observation plane signed distance",
+        description="Signed distance of the observation plane along ``plane_axis`` "
+        "w.r.t. ``local_origin``",
+    )
+
+    x: ObsGridArray = pydantic.Field(
+        ...,
+        title="Local x observation coordinates",
+        description="Local x observation coordinates w.r.t. ``local_origin`` and ``plane_axis``. "
+        "When ``plane_axis`` is 0, this corresponds to the global y axis. "
+        "When ``plane_axis`` is 1, this corresponds to the global x axis. "
+        "When ``plane_axis`` is 2, this corresponds to the global x axis. ",
+        units=MICROMETER,
+    )
+
+    y: ObsGridArray = pydantic.Field(
+        ...,
+        title="Local y observation coordinates",
+        description="Local y observation coordinates w.r.t. ``local_origin`` and ``plane_axis``. "
+        "When ``plane_axis`` is 0, this corresponds to the global z axis. "
+        "When ``plane_axis`` is 1, this corresponds to the global z axis. "
+        "When ``plane_axis`` is 2, this corresponds to the global y axis. ",
+        units=MICROMETER,
+    )
+
+    def storage_size(self, num_cells: int, tmesh: ArrayLike[float, 1]) -> int:
```

Comment:
Let's put the docstring
```py
"""Size of monitor storage given the number of points after discretization."""
```
under all of these `storage_size` methods as part of this PR.